eahub/base/views.py

```diff
@@ -130,6 +140,48 @@ class RobotsTxtView(base.TemplateView):
     template_name = "robots.txt"
     content_type = "text/plain; charset=utf-8"
 
+class ReportAbuse:
+    def __init__(self, reportee, type):
+        self.reportee = reportee
+        self.type = type
+
+    def send(self, request, form):
+        reasons = form.cleaned_data
+        subject = "EA {0} reported as abuse: {1}".format(self.type, self.reportee.name)
+        message = render_to_string('emails/report_{}_abuse.txt'.format(self.type), {
+            'profile_name': self.reportee.name,
+            'profile_url': "https://{0}/profile/{1}".format(get_current_site(request).domain,self.reportee.slug),
+            'reasons': ', '.join(reasons)
+        })
+        send_mail(subject, message, settings.DEFAULT_FROM_EMAIL, recipient_list=settings.LEAN_MANAGERS)
+        messages.success(
+            request,
+            ''' Thank you, we have received your report. Our admin team will send you an email once they have looked into it. ''',
+        )
+        return redirect('/{0}/{1}'.format(self.type,self.reportee.slug))
+
+class ReportProfileAbuseView(FormView):
+    template_name = 'eahub/report_abuse.html'
+    form_class = ReportAbuseForm
+
+    def form_valid(self, form):
+        report_abuse = self.get_report_abuse()
+        return report_abuse.send(self.request, form)
+
+    def get_report_abuse(self):
+        return ReportAbuse(self.profile(), 'profile')
+
+    def profile(self):
+        return Profile.objects.get(slug=self.kwargs['slug'])
+
+
+class ReportGroupAbuseView(ReportProfileAbuseView):
```

Comment:
These two classes are siblings, so one shouldn't extend the other; they should both extend a common ancestor. That way, you can put the ancestor in `base` and the app-specific ones in those apps.